group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/RangeAssignor.java

```diff
@@ -90,17 +91,29 @@ private Map<Uuid, List<String>> membersPerTopic(final AssignmentSpec assignmentS
         Map<Uuid, List<String>> membersPerTopic = new HashMap<>();
         Map<String, AssignmentMemberSpec> membersData = assignmentSpec.members();
 
-        membersData.forEach((memberId, memberMetadata) -> {
-            Collection<Uuid> topics = memberMetadata.subscribedTopicIds();
+        if (assignmentSpec.groupSubscriptionModel().equals(HOMOGENEOUS)) {
+            List<String> allMembers = new ArrayList<>(membersData.keySet());
```

Comment:
I wonder if we could change the return type of the method from `Map<Uuid, List<String>>` to `Map<Uuid, Collection<String>>` and avoid this copy here. It seems possible because we only iterate over the member ids later on. This could be a nice performance improvement too while we are here.

Reply:
Thanks! Made the change